kuka_rsi_hw_interface/include/kuka_rsi_hw_interface/kuka_hardware_interface.hpp

```diff
@@ -28,57 +28,42 @@
 #include "rclcpp_lifecycle/lifecycle_node.hpp"
 #include "sensor_msgs/msg/joint_state.hpp"
 
-
 namespace kuka_rsi_hw_interface
 {
-class KukaHardwareInterface
-{
-public:
-  explicit KukaHardwareInterface(rclcpp_lifecycle::LifecycleNode::SharedPtr robot_control_node);
-
-  void start();
-  void stop();
-  void configure();
-  void cleanup();
-  bool read();
-  void write();
-
-  const bool & isActive() const;
-
-private:
-  void commandReceivedCallback(sensor_msgs::msg::JointState::SharedPtr msg);
-
-  unsigned int n_dof_ = 6;
-  bool is_active_ = false;
-
-  std::vector<std::string> joint_names_ = std::vector<std::string>(6);
-
-  RSIState rsi_state_;
-  RSICommand rsi_command_;
-  std::vector<double> initial_joint_pos_ = std::vector<double>(6, 0.0);
-  std::vector<double> joint_pos_correction_deg_ = std::vector<double>(6, 0.0);
-  uint64_t ipoc_ = 0;
-
-  std::unique_ptr<UDPServer> server_;
-  std::string local_host_ = "";
-  int local_port_ = 0;
-  std::string in_buffer_;
-  std::string out_buffer_;
-
-  rclcpp_lifecycle::LifecycleNode::SharedPtr control_node_;
-  rclcpp_lifecycle::LifecyclePublisher<sensor_msgs::msg::JointState>::SharedPtr
-    joint_state_publisher_;
-  rclcpp::Subscription<sensor_msgs::msg::JointState>::SharedPtr joint_command_subscription_;
-  rclcpp::CallbackGroup::SharedPtr cbg_;
-  sensor_msgs::msg::JointState::SharedPtr joint_command_msg_;
-  sensor_msgs::msg::JointState joint_state_msg_;
-
-  std::mutex m_;
-  std::condition_variable cv_;
-
-  static constexpr double R2D = 180 / M_PI;
-  static constexpr double D2R = M_PI / 180;
-};
-}  // namespace kuka_rsi_hw_interface
-
-#endif  // KUKA_RSI_HW_INTERFACE__KUKA_HARDWARE_INTERFACE_HPP_
+  class KukaHardwareInterface
+  {
+  public:
+    explicit KukaHardwareInterface(const std::string &rsi_ip_address, int rsi_port, unsigned int n_dof);
+
+    void start(std::vector<double> &joint_state_msg_position,
+               std::vector<double> &joint_command_msg_position,
+               std::vector<double> &initial_joint_position);
+    void stop(const std::vector<double> &joint_pos_correction_deg_);
+    bool read(std::vector<double> &joint_state_msg_position);
+    bool write(const std::vector<double> &joint_pos_correction_deg_);
+
+    bool isActive() const;
+
+  private:
+    unsigned int n_dof_ = 6;
+    bool is_active_ = false;
+
+    std::vector<std::string> joint_names_ = std::vector<std::string>(6);
```

Comment:
use n_dof_ instead of 6

Reply:
Actually this is not used, and probably not even needed. Deleted.